balancer-js/src/modules/zaps/bbausd2-migrations/stables.ts

```diff
@@ -0,0 +1,246 @@
+   * @param {string}                    userAddress User address.
+   * @param {{string, string, string}}  from Pool info being migrated from
+   * @param {{string, string, string}}  to Pool info being migrated to
+   * @param {string}                    bptIn Amount of BPT tokens to migrate.
+   * @param {string}                    minBptOut Minimum of expected BPT out ot the migration flow.
+   * @param {boolean}                   staked Indicates whether tokens are initially staked or not.
+   * @param {string[]}                  tokens Token addresses. Array must have the same length and order as tokens in pool being migrated from. Refer to [getPoolTokens](https://github.com/balancer-labs/balancer-v2-monorepo/blob/master/pkg/interfaces/contracts/vault/IVault.sol#L334).
+   * @param {string}                    authorisation Encoded authorisation call.
+   * @returns Migration transaction request ready to send with signer.sendTransaction
+   */
+  calldata(
+    userAddress: string,
+    from: { id: string; address: string; gauge?: string },
+    to: { id: string; address: string; gauge?: string },
+    bptIn: string,
+    minBptOut: string = MaxInt256.toString(),
+    staked: boolean,
+    tokens: string[],
+    authorisation?: string
+  ): {
+    to: string;
+    data: string;
+  } {
+    const relayer = this.addresses.relayer;
+    let calls: string[] = authorisation
+      ? [this.buildSetRelayerApproval(authorisation)]
+      : [];
+
+    if (staked && (from.gauge == undefined || to.gauge == undefined))
+      throw new Error(
+        'Staked flow migration requires gauge addresses to be provided'
+      );
+
+    if (staked) {
+      calls = [
+        ...calls,
+        this.buildWithdraw(userAddress, bptIn, from.gauge as string),
+        this.buildExit(from.id, relayer, relayer, bptIn, tokens),
+        this.buildSwap(minBptOut, relayer, to.id, to.address, tokens),
+        this.buildDeposit(userAddress, to.gauge as string),
+      ];
+    } else {
+      calls = [
+        ...calls,
+        this.buildExit(from.id, userAddress, relayer, bptIn, tokens),
+        this.buildSwap(minBptOut, userAddress, to.id, to.address, tokens),
+      ];
+    }
+
+    const callData = balancerRelayerInterface.encodeFunctionData('multicall', [
+      calls,
+    ]);
+
+    return {
+      to: this.addresses.relayer,
+      data: callData,
+    };
+  }
+
+  /**
+   * Encodes exitPool call data.
+   * Exit stable pool proportionally to underlying stables. Exits to relayer.
+   * Outputreferences are used to store exit amounts for next transaction.
+   *
+   * @param {string}    poolId Pool id.
+   * @param {string}    sender Sender address.
+   * @param {string}    recipient Recipient address.
+   * @param {string}    amount Amount of BPT to exit with.
+   * @param {string[]}  tokens Token addresses to exit to.
+   * @returns Encoded exitPool call. Output references.
+   */
+  buildExit(
+    poolId: string,
+    sender: string,
+    recipient: string,
```

Comment:
As we're always exiting to Relayer should we remove this parameter to make it more defined?

Reply:
👍 ✅